src/events/ui_events.rs

```diff
@@ -0,0 +1,147 @@
+use std::{sync::Arc, task::Poll};
+
+use crate::{
+    chat::ChatId,
+    contact::{Contact, ContactId},
+    context::Context,
+    EventType,
+};
+use async_channel::{self as channel, Receiver, Sender};
+use channel::{bounded, TrySendError};
+use futures::{executor::block_on, Future};
+use tokio::sync::RwLock;
+
+/// order or content of chatlist changes (chat ids, not the actual chatlist item)
+pub(crate) fn emit_chatlist_changed(context: &Context) {
+    context.emit_event(EventType::UIChatListChanged);
+}
+
+/// Chatlist item of a specific chat changed
+pub(crate) fn emit_chatlist_item_changed(context: &Context, chat_id: ChatId) {
+    context.emit_event(EventType::UIChatListItemChanged {
+        chat_id: Some(chat_id),
+    });
+}
+
+#[allow(unused)]
+/// Used when you don't know which chatlist items changed, this reloads all cached chatlist items in the UI
+/// note(treefit): This is not used right now, but I know there will be a point where someone wants it
```

Comment:
I bet we'll use this "escape hatch" at some point when we quickly need to update everything and do not have the chatid at hand. like now I changed contact name to use it instead of running the expensive db query to find the real chat ids.